tests/tests.rs

```diff
@@ -211,3 +370,35 @@ fn test_provider_registry() {
         ]
     )
 }
+
+#[test]
+fn free_rpc_auth() {
```

Comment:
@rvanasa nit: a more descriptive test name would be useful, e.g., `should_query_gas_price_from_free_rpc_provider`

Reply:
I had to read your descriptive test name a few times to get it...   :smile: 
Can we simplify it somehow? E.g., `free_prc_provider_should_provide_gas_price` (or something even simpler)?

Reply:
Totally ok with other names, this was just a suggestion. The main point is that it should be clear just by reading the test name what's being test/what is the goal of the test.

Reply:
Yes, for sure. Thanks for pointing this out. I originally had a better name and then tried to shorten it, which made it less clear. Updated both of the existing test names based on this feedback.